src/backend/commands/cluster.c

```diff
@@ -734,8 +734,14 @@ make_new_heap(Oid OIDOldHeap, Oid NewTableSpace, Oid NewAccessMethod,
 	 * a shared rel.  However, we do make the new heap mapped if the source is
 	 * mapped.  This simplifies swap_relation_files, and is absolutely
 	 * necessary for rebuilding pg_class, for reasons explained there.
+	 *
+	 * We also must ensure that these temp tables are properly named in TSQL
+	 * so that the metadata is properly cleaned up after in this function.
 	 */
-	snprintf(NewHeapName, sizeof(NewHeapName), "pg_temp_%u", OIDOldHeap);
+	if (sql_dialect == SQL_DIALECT_TSQL && relpersistence == RELPERSISTENCE_TEMP && get_ENR_withoid(currentQueryEnv, OIDOldHeap, ENR_TSQL_TEMP))
```

Comment:
can we be sure this change on its own has no side effects? Since it was packaged as part of a bigger change in rel16, and Rel16 also has other commits that don't exist on Rel15.

Reply:
As far as I can tell, there are no side effects of this change. All JDBC tests pass, and it still has the intended benefit of allowing ALTER COLUMN <type> queries to go through. In terms of static analysis, I think this looks like a safe change to make - all we are doing is using a slightly different name for the generated temp table, which is still valid for both temp and permanent table naming conventions anyways.